pinot-core/src/main/java/com/linkedin/pinot/core/query/pruner/PartitionSegmentPruner.java

```diff
@@ -99,8 +103,41 @@ public boolean pruneSegment(FilterQueryTree filterQueryTree, Map<String, ColumnM
       return false;
     }
 
-    Comparable value = getValue(filterQueryTree.getValue().get(0), metadata.getDataType());
     PartitionFunction partitionFunction = metadata.getPartitionFunction();
+
+    if (operator == FilterOperator.EQUALITY) {
+      return pruneEquality(filterQueryTree, metadata, partitionRanges, partitionFunction);
+    }
+
+    return pruneIn(filterQueryTree, metadata, partitionRanges, partitionFunction);
+  }
+
+  private boolean pruneIn(FilterQueryTree filterQueryTree, ColumnMetadata metadata, List<IntRange> partitionRanges,
```

Comment:
I am a bit uncomfortable with having to create these string garbages per query per segment. For a recent high throughput case, I found this to be creating a lot of garbage, and I had modified the pruner to take an already created filterQueryTree as opposed to creating it over and over again per segment. I fear this change will cause the same issue.